web/client-api/src/main/java/io/deephaven/web/client/api/JsTable.java

```diff
@@ -1715,7 +1714,7 @@ public void setSize(double s) {
         this.size = s;
 
         TableViewportSubscription subscription = subscriptions.get(getHandle());
-        if (changed && (subscription == null || subscription.getStatus() == TableViewportSubscription.Status.DONE)) {
+        if (changed && (subscription == null || !subscription.hasValidSize())) {
```

Comment:
Guessing we don't need to worry about the `STARTING` state here? Or shouldn't this just be checking `subscription.isClose()` (as it was just checking against == `DONE` before).

Reply:
If the subscription is in the STARTING state, we havent yet received a definitive size for the subscription, and so we should continue to use the "other" way that we get sizes. That is, the server sends ExportedTableUpdateMessage values that show the new size of the table whenever it changes, but those are sometimes faster than the update interval for the subscription, so technically using the subscription means we might skip some size changes (the purpose of this `if`).

However, that's a good thing. We don't want tell the UI that the table size is changing in a way that is actually incompatible with the subscription's own contents.

The old behavior here was to say "if you have started a subscription, but the first message hasnt arrived from the server, don't worry about the size changes, you'll get an accurate picture when the snapshot gets here". The bug report that led to the PR pointed out that sometimes it can be seconds (or minutes) before the first snapshot gets here, and we need to have accurate information before, during, and after.

The actual fix for the patch is to say "if the subscription is actively changing, assume the old value is still good, and the next subscription message is the best way to get updates". Formerly, we would actually create a new subscription to the server when the viewport was adjusted, so "before" that subscription was ready also encompassed "during, but temporarily invalid". Now, that "temporarily invalid" state doesn't require a new subscription, so we can treat the "before" state as "not actually ready to hand the reins to the subscription".